### ISSUE TITLE
Typescript

### ISSUE THREAD

**Issue by theoparis** [Irrelon/irrelon-path#12]
Is it possible to have typescript declarations implemented in this project? I really like that its up to date (9 days ago) and that there's a lot of options/customizablilty. However, it doesn't seem like its ts-compatible :/

** Can you make these changes in a PR **
I could make a PR if you need me to.

**Comment by Irrelon** [comment]
Hi there. Most of the functions are typed already via JSDoc. Are you able to read types from that?

**Comment by theoparis** [comment]
Yes, however my ide complains that it does have declaration types. Maybe there's a way to generate them from jsdoc?

The error:
```
Could not find a declaration file for module '@irrelon/path'. 
```

**Comment by theoparis** [comment]
I think I found a way to do this, I am going to create pr if it works.

**Comment by theoparis** [comment]
Also, shouldn't the main field in package.json point to dist instead of src?

**Comment by Irrelon** [comment]
Cool let me know how it goes.

**Comment by Irrelon** [comment]
As far as dist vs src, dist is for browser compat. Src works in node and any projects that have Babel or es6 capability. Requiring the package from node and having it load from dist would include a bunch of module compatibility wrapper code that is totally not required. That’s the rationale. I think there is a browser field that points to dist if I remember correctly.

**Comment by theoparis** [comment]
Closed by #13
